### DISCUSSION TITLE
Comparing `model.dsl` to the latest deployed authorization model

### DISCUSSION THREAD

**Discussion by vhpoet** [openfga/community#68] [Q&A]
I have code that will take `model.dsl` and write it to the store, but I only want to do it when `model.dsl` is different from what's already in the store. Is there an elegant way of doing this?

Current approach is to: 
- Turn `model.dsl` into json format.
- Load latest model from the store.
- Compare.
 
A bit of weirdness when comparing: I get this `"metadata": null` thing from the loaded model that I don't have in the local modal. See below.

```
[
  {
    "metadata": null,
    "relations": {
      "can_update_user": {
        "computedUserset": {
          "object": "",
          "relation": "admin"
        }
      },
    },
    "type": "customer"
  }
]
```
vs
```
[
  {
    "relations": {
      "can_update_user": {
        "computedUserset": {
          "object": "",
          "relation": "admin"
        }
      },
    },
    "type": "customer"
  }
]
```

**Comment by rhamzeh** [comment]
That's a great question @vhpoet - internally what we were doing was exactly what you said, but we strip out some fields.

FYI - Metadata will be important in v1.1 of the DSL as well as a schema version field.

We can consider exposing a function in the syntax transformer that handles checking whether two versions are the same.

One question I have. Assume later on we allowed you to add comments to the JSON/DSL, would you consider changes to the comments as a divergence that would cause a write? (We'd say yes, but curious if you'd think the same)
